bootique/src/main/java/io/bootique/meta/application/CommandMetadata.java

```diff
@@ -83,9 +87,40 @@ public String getDescription() {
      * @return option representation of this command.
      */
     public OptionMetadata asOption() {
-        // TODO: cache the value?
-        // using getters instead of vars ; some getters have logic
-        return OptionMetadata.builder(getName()).shortName(getShortName()).description(getDescription()).build();
+        OptionValueCardinality cardinality = getValueCardinality();
+        String valueName = getValueName();
+
+        Consumer<OptionMetadata.Builder> optionMetadataConsumer = createOptionsConsumer(cardinality, valueName);
+
+        OptionMetadata.Builder optionBuilder = OptionMetadata.builder(getName())
+                .shortName(getShortName())
+                .description(getDescription())
+                .applyCardinality(optionMetadataConsumer);
+
+        if (defaultValue != null) {
+            optionBuilder.valueOptionalWithDefault(valueName, defaultValue);
+        }
+
+        return optionBuilder.build();
+    }
+
+    private static Consumer<OptionMetadata.Builder> createOptionsConsumer(OptionValueCardinality cardinality, String valueName) {
```

Comment:
Per my comment on `OptionMetadata`, let's simplify this, removing the "consumer" indirection.